lib/SIL/IR/AbstractionPattern.cpp

```diff
@@ -61,6 +61,9 @@ static const clang::Type *getClangType(const clang::Decl *decl) {
     return valueDecl->getType().getTypePtr();
   }
 
+  if (auto recordDecl = dyn_cast<clang::RecordDecl>(decl))
+    return recordDecl->getTypeForDecl();
+
```

Comment:
The clang declaration here should always be a value decl.  If someone's passing in a record, that should be fixed to use a clang type pattern.